src/seedu/addressbook/data/person/AddressComponent.java

```diff
@@ -0,0 +1,13 @@
+package seedu.addressbook.data.person;
+
+abstract class AddressComponent {
```

Comment:
Since we have not covered inheritance yet it would have been ok just to use separate classes. However, this is a good try in implementing inheritance. 

You could have common methods such as toString().hashCode(), equals() also implemented here.